vm/src/exceptions.rs

```diff
@@ -1408,4 +1389,218 @@ pub(super) mod types {
+                            #[cfg(target_os = "wasi")]
+                            const ESRCH: i32 = wasmer_wasi::types::__WASI_ESRCH as i32;
+
+                            #[cfg(not(any(windows, target_os = "wasi")))]
+                            const ETIMEDOUT: i32 = libc::ETIMEDOUT;
+                            #[cfg(windows)]
+                            const ETIMEDOUT: i32 = winapi::shared::winerror::WSAETIMEDOUT as i32;
+                            #[cfg(target_os = "wasi")]
+                            const ETIMEDOUT: i32 = wasmer_wasi::types::__WASI_ETIMEDOUT as i32;
+
+                            let excs = &vm.ctx.exceptions;
+                            let error = match errno {
+                                EWOULDBLOCK => Some(excs.blocking_io_error.clone()),
+                                EALREADY => Some(excs.blocking_io_error.clone()),
+                                EINPROGRESS => Some(excs.blocking_io_error.clone()),
+                                EPIPE => Some(excs.broken_pipe_error.clone()),
+                                ESHUTDOWN => Some(excs.broken_pipe_error.clone()),
+                                ECHILD => Some(excs.child_process_error.clone()),
+                                ECONNABORTED => Some(excs.connection_aborted_error.clone()),
+                                ECONNREFUSED => Some(excs.connection_refused_error.clone()),
+                                ECONNRESET => Some(excs.connection_reset_error.clone()),
+                                EEXIST => Some(excs.file_exists_error.clone()),
+                                ENOENT => Some(excs.file_not_found_error.clone()),
+                                EISDIR => Some(excs.is_a_directory_error.clone()),
+                                ENOTDIR => Some(excs.not_a_directory_error.clone()),
+                                EINTR => Some(excs.interrupted_error.clone()),
+                                EACCES => Some(excs.permission_error.clone()),
+                                EPERM => Some(excs.permission_error.clone()),
+                                ESRCH => Some(excs.process_lookup_error.clone()),
+                                ETIMEDOUT => Some(excs.timeout_error.clone()),
+                                _ => None,
+                            };
+
+                            if error.is_some() {
+                                Some(invoke(error?, args.to_vec(), vm))
+                            } else {
+                                None
+                            }
+                        }
+                        Err(_) => None,
+                    };
+
+                    error
+                }
+                None => None,
+            };
+
+            error
+        } else {
+            None
+        }
+    }
+
+    fn base_exception_new(cls: PyTypeRef, args: FuncArgs, vm: &VirtualMachine) -> PyResult {
+        PyBaseException::slot_new(cls, args, vm)
+    }
```

Comment:
if these methods are not required to be moved, keep them on the original place. they will be put in `impl` later once we revise `pyexception` macro

Reply:
They are used in define_macro.

Reply:
I agree it is used. I don't see a good reason why it need to be moved to here.

Reply:
I agree it is used. I don't see a good reason why it need to be moved to here. I checked out the code and tried to compile it without moving this function. I couldn't see any problem.

Reply:
I see. I moved functions just for distinguishing functions and macros. For readability, if you want not moving the functions I can revert that.